dbms/src/Functions/FunctionsJson.h

```diff
@@ -105,197 +112,221 @@ class FunctionJsonExtract : public IFunction
     void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
     {
         size_t rows = block.rows();
+        auto & res_col = block.getByPosition(result).column;
         /// First check if JsonObject is only null, if so, result is null, no need to parse path exprs even if path is
         const auto * json_column = block.getByPosition(arguments[0]).column.get();
-        bool all_null = false;
-        bool const_json = false;
-        const ColumnString * source_data_column_ptr;
-        const ColumnNullable * source_nullable_column_ptr = nullptr;
         if unlikely (json_column->onlyNull())
         {
-            block.getByPosition(result).column = block.getByPosition(result).type->createColumnConst(rows, Null());
+            res_col = block.getByPosition(result).type->createColumnConst(rows, Null());
             return;
         }
-        else if (const auto * const_nullable_col = checkAndGetColumnConst<ColumnNullable>(json_column))
-        {
-            const_json = true;
-            json_column = const_nullable_col->getDataColumnPtr().get();
-        }
-        else if (const auto * const_col = checkAndGetColumnConst<ColumnString>(json_column))
+        if (json_column->isColumnNullable())
         {
-            const_json = true;
-            json_column = const_col->getDataColumnPtr().get();
-        }
-
-        if (const auto * nullable_col = checkAndGetColumn<ColumnNullable>(json_column))
-        {
-            source_nullable_column_ptr = nullable_col;
-            all_null = isColumnOnlyNull(nullable_col); /// changes
-            source_data_column_ptr = checkAndGetColumn<ColumnString>(nullable_col->getNestedColumnPtr().get());
-            if unlikely (!source_data_column_ptr)
-                throw Exception(
-                    fmt::format("Illegal column {} of argument of function {}", json_column->getName(), getName()),
-                    ErrorCodes::ILLEGAL_COLUMN);
-        }
-        else if (const auto * string_col = checkAndGetColumn<ColumnString>(json_column))
-        {
-            source_data_column_ptr = string_col;
-        }
-        else
-        {
-            throw Exception(
-                fmt::format("Illegal column {} of argument of function {}", json_column->getName(), getName()),
-                ErrorCodes::ILLEGAL_COLUMN);
+            const auto & column_nullable = static_cast<const ColumnNullable &>(*json_column);
+            const auto & null_map = column_nullable.getNullMapData();
+            auto null_count = countBytesInFilter(null_map.data(), null_map.size());
+            if (null_count == rows)
+            {
+                res_col = block.getByPosition(result).type->createColumnConst(rows, Null());
```

Comment:
I think even if `null_count == rows` is true, it does not imply that current column only contains null? Since this is only one block of this column, some other block may contain not null value.

Reply:
ok, these code removed.